### ISSUE TITLE
New error messages for `alt`-less `img` with ARIA attributes sometimes cause problems

### ISSUE THREAD

**Issue by dd8** [validator/validator#1000]
The new alt-less img messages can cause problems in certain scenarios:
https://github.com/validator/validator/commit/3a010b40d68877faf157edd5fb9bfefb3a3cd635

1) An img element with no alt attribute must not have any aria-* attributes other than aria-hidden.
This encourages the user to remove aria-label (and the accessible name) in the following code. This changes voicing in a screen reader from "Image, Fish in a tank" to 'Image one thousand two hundred and thirty four dot PNG"
```
<img aria-label='Fish in a tank' src='1234.png' />
```

2) An img element with no alt attribute must not have a role attribute.
This encourages the user to remove `role=none`  or `role=presentation` (which have same effect as `alt=''`) in the following code. With `role=none` present the image is ignored in a screen reader. Removing the role causes the image to be voiced in a screen reader as 'Image one thousand two hundred and thirty four dot PNG"
```
<img role='none' src='1234.png' />
```

**Comment by sideshowbarker** [comment]
Yes, I see what you mean.

What do think of the following replacements?

> An img element with any aria-* attributes other than aria-hidden must also have an alt attribute.

> An img element with a role attribute must also have an alt attribute.

**Comment by dd8** [comment]
Stepping through what a user might do to fix this and the messages they get along the way:

## Example 1
```
<img aria-label='Fish in a tank' src='1234.png' />
```
Will now report
> An `img` element with any `aria-*` attributes other than `aria-hidden` must also have an `alt` attribute

Path 1 - Author fixes by adding `alt='' `
```
<img aria-label='Fish in a tank' src='1234.png' alt='' />
```
Reports no errors: `alt=''` would usually mark image as presentational, but `aria-label` triggers presentational role conflict resolution, so is read as "Image, Fish in a tank"

Path 2 - Author fixes by adding `alt='Fish in a tank'`
```
<img aria-label='Fish in a tank' src='1234.png' alt='Fish in a tank' />
```
Reports no errors, and is read as "Image, Fish in a tank"

So should end up in a better place with
> An `img` element with any `aria-*` attributes other than `aria-hidden` must also have an `alt` attribute

## Example 2
```
<img role='none' src='1234.png' />
```
> An `img` element with any `aria-*` attributes other than `aria-hidden` must also have an `alt` attribute

Path 1 - Author fixes by adding `alt=''`
```
<img role='none' src='1234.png' alt='' />
```
Reports no errors, and is still ignored as intended

Path 2 - Author fixes by adding `alt='Fish in a tank'`
```
<img role='none' src='1234.png' alt='Fish in a tank' />
```
Reports 
>Bad value `none` for attribute role on element `img`

It's not clear why role is a bad value here (which I think was one of the original problems in https://github.com/validator/validator/issues/955). 

The second message is also better although path 2 leads to a confusing message (but that happens already)
> An `img` element with any `aria-*` attributes other than `aria-hidden` must also have an `alt` attribute

**Comment by sideshowbarker** [comment]
Changed the messages along the lines I described in https://github.com/validator/validator/issues/1000#issuecomment-665412116, and deployed.
